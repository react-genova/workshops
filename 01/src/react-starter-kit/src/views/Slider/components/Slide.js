//@flow
import * as React from 'react';
import styled from 'styled-components';

import SlideHeader from './SlideHeader';
import SlideFooter from './SlideFooter';

const Container = styled.div`
    display: flex;
    min-height: 100vh;
    flex-direction: column;
`;

const SlideBody = styled.div`
    display: flex;
    justify-content: center;
    flex: 1;
    background-color: #dae6f1;
    margin-top: 0.2em;
    margin-bottom: 0.2em;
`;

type Props = {
    title:string,
    description:string,
    children?:React.Node,
};

const Slide = ({title, description, children}:Props) => (
    <Container>
        <SlideHeader title={title} description={description} />
        <SlideBody>{children}</SlideBody>
        <SlideFooter />
    </Container>
);

export default Slide;