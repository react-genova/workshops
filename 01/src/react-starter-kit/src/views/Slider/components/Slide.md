### Welcome slide ###

```js
<Slide title="Welcome" description="to madness">
    <SlideWelcome />
</Slide>
```

### All you need is L.O.V.S. slide ###

```js
<Slide title="All you need is L.O.V.S." description="List Of Valuable Suggestions">
    <SlideLOVS />
</Slide>
```

### Conclusions slide ###

```js
<Slide title="Conclusions" description="Yeah, interesting, so in short?">
    <SlideConclusions />
</Slide>
```