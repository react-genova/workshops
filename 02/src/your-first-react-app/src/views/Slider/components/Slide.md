### Welcome slide ###

```js
<Slide title="Welcome" description="to madness">

</Slide>
```
