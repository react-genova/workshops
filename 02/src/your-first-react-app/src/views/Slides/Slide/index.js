//@flow
export {default as Container} from './components/Container';
export {default as Important} from './components/Important';