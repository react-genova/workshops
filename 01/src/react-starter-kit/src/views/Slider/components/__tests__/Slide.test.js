//@flow
import React from 'react';
import Slide from '../Slide';
import { shallow } from 'enzyme';

describe('Slide component', () => {

    it('simple header snapshot', () => {
        const component = shallow(
            <Slide title="Welcome" description="to madness">
                <div>WELCOME</div>
            </Slide>);
        expect(component).toMatchSnapshot();
    });

});