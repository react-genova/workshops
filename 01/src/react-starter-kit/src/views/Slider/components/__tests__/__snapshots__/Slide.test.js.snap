// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide component simple header snapshot 1`] = `
<styled.div>
  <SlideHeader
    description="to madness"
    title="Welcome"
  />
  <styled.div>
    <div>
      WELCOME
    </div>
  </styled.div>
  <SlideFooter />
</styled.div>
`;
